src/qtils/macro/common.hpp

```diff
@@ -0,0 +1,15 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#define IDENTITY(a) a
+
+#define CONCAT__(a, b) a##b
+#define CONCAT_(a, b) CONCAT__(a, b)
+#define CONCAT(a, b) CONCAT_(IDENTITY(a), IDENTITY(b))
+
+#define QTILS_UNIQUE_NAME(prefix) CONCAT(prefix, __COUNTER__)
```

Comment:
Are `IDENTITY` and `CONCAT` used somewhere else?
```suggestion
#define _QTILS_UNIQUE_NAME_2(x, y) x##y
#define _QTILS_UNIQUE_NAME(prefix, counter) _QTILS_UNIQUE_NAME_2(prefix, counter)
#define QTILS_UNIQUE_NAME_PREFIX(prefix) _QTILS_UNIQUE_NAME(prefix, __COUNTER__)
#define QTILS_UNIQUE_NAME QTILS_UNIQUE_NAME_PREFIX(qtils_unique_name_)
```

Reply:
Not yet, but they are very commonly used macros so I think it's better to keep them separate. Maybe add a prefix though.